cam/cam.go

```diff
@@ -400,15 +368,26 @@ func (vs *videostore) processFrames(ctx context.Context) {
 				vs.logger.Debug("latest frame is not available yet")
 				continue
 			}
-			encoded, pts, dts, err := vs.enc.encode(*latestFrame)
+			result, err := vs.enc.encode(*latestFrame)
 			if err != nil {
 				vs.logger.Error("failed to encode frame", err)
-				return
+				continue
 			}
-			err = vs.seg.writeEncodedFrame(encoded, pts, dts)
+			if result.frameDimsChanged {
+				vs.logger.Info("reinitializing segmenter due to encoder refresh")
+				err = vs.seg.initialize(vs.enc.codecCtx)
+				if err != nil {
+					vs.logger.Error("failed to reinitialize segmenter", err)
```

Comment:
how do you feel about making these errors debug level? Are they edge case enough to warrant that rather than spamming the user?

Reply:
Sounds good. We hopefully will not be getting into a state where encoder and/or segmenter are broken and we are spamming in a loop, but changed to debug logs to be safe.